src/puppet-web/wechaty-bro.js

```diff
@@ -378,6 +379,19 @@
     })
     return true
   }
+
+  function hookProcess() {
+    var chatFactory = WechatyBro.glue.chatFactory
+    var rootScope = WechatyBro.glue.rootScope
+    // hook chatFactory._recalledMsgProcess, resolve emit RECALLED type msg
+    chatFactory.__recalledMsgProcess = chatFactory._recalledMsgProcess
+    chatFactory._recalledMsgProcess = function (msg) {
+      var _this = this
+      _this.__recalledMsgProcess(msg)
```

Comment:
How about use `chatFactory` at here instead of `this`?

Like: `chatFactory.__recalledMsgProcess` and `__recalledMsgProcess.addChatMessage`